kernel/classes/datatypes/ezuser/ezuser.php

```diff
@@ -1804,6 +1816,19 @@ static function createHash( $user, $password, $site, $type, $hash = false )
         {
             $str = $password;
         }
+        else if ( $type == self::PASSWORD_BCRYPT )
+        {
+            if ( $hash ) {
+                if ( password_verify( $password, $hash ) ) {
+                    return $hash;
+                } else {
+                    return false;
+                }
+            } else {
+                $str = password_hash( $password, PASSWORD_BCRYPT );
+            }
+            $str = $password;
```

Comment:
This here is wrong. This means that unhashed password will be returned from this method, always.